specifyweb/frontend/js_src/lib/components/Interactions/PrepDialogRow.tsx

```diff
@@ -79,8 +83,19 @@ export function PrepDialogRow({
             max={preparation.available}
             min={0}
             title={interactionsText.selectedAmount()}
-            value={selected}
-            onValueChange={handleChange}
+            value={
```

Comment:
This (very complicated) logic only affects what's visible in the field, not what's going to be sent to the server when user submits the form

Reply:
You shouldn't have to change anything in this file.
Instead, in the `onValueChange` for the bulk value input field, call setSelected to change how many are selected in each